crates/bytecode/src/legacy/analysis.rs

```diff
@@ -39,7 +50,12 @@ pub fn analyze_legacy(bytecode: Bytes) -> (JumpTable, Bytes) {
         }
     }
 
-    let padding = (iterator as usize) - (end as usize) + (opcode != opcode::STOP) as usize;
+    let overflow_padding = (iterator as usize) - (end as usize);
+
+    let stop_padding = if !is_terminating_opcode(opcode) { 1 } else { 0 };
```

Comment:
```suggestion
    let stop_padding = OpCode::info_by_op(opcode).map(|o| !o.is_terminating() as usize).unwrap_or(1);
```

can replace the function.